File: Entry.js
class Entry {
    constructor(date=null, dist=null, loc=null, purp=null) {
        this.date = date;
        this.dist = dist;
        this.loc = loc;
        this.purp = purp;
    }

    editDate(new_date) {this.date = new_date}

    editDist(new_dist) {this.dist = new_dist}

    editLoc(new_loc) {this.loc = new_loc}

    editPurp(new_purp) {this.purp = new_purp}
}

File: Archive/Entry.py
class Entry:
    def __init__(self, date, dist, loc, purp):
        self.date = date
        self.dist = dist
        self.loc = loc
        self.purp = purp

    def edit_date(self, new_date):
        self.date = new_date

    def edit_dist(self, new_dist):
        self.dist = new_dist

    def edit_loc(self, new_loc):
        self.loc = new_loc

    def edit_purp(self, new_purp):
        self.purp = new_purp

    def __str__(self):
        return str(self.date[0]) + "/" + str(self.date[1]) + "/" + str(self.date[2]) + " | " + str(self.dist) + " | " + self.loc + " | " + self.purp